static/app/components/events/interfaces/spans/traceErrorList.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import flatten from 'lodash/flatten';
+import groupBy from 'lodash/groupBy';
+
+import List from 'sentry/components/list';
+import ListItem from 'sentry/components/list/listItem';
+import {t, tn} from 'sentry/locale';
+import {TraceError} from 'sentry/utils/performance/quickTrace/types';
+
+import {ParsedTraceType, SpanType} from './types';
+
+interface TraceErrorListProps {
+  errors: TraceError[];
+  onClickSpan: (event: React.MouseEvent, spanId: SpanType['span_id']) => void;
+  trace: ParsedTraceType;
+}
+
+function TraceErrorList({trace, errors, onClickSpan}: TraceErrorListProps) {
+  return (
+    <List symbol="bullet" data-test-id="trace-error-list">
+      {flatten(
+        Object.entries(groupBy(errors, 'span')).map(([spanId, spanErrors]) => {
+          const span = findSpanById(trace, spanId);
+
+          return Object.entries(groupBy(spanErrors, 'level')).map(
+            ([level, spanLevelErrors]) => (
+              <ListItem key={`${spanId}-${level}`}>
+                {t('%d', spanLevelErrors.length)} {` ${level === 'error' ? '' : level} `}
+                {tn('error', 'errors', spanLevelErrors.length)}
+                {span ? (
+                  <React.Fragment>
+                    {' in '}
+                    <ErrorLink onClick={event => onClickSpan(event, spanId)}>
+                      {span.op}
+                    </ErrorLink>
+                  </React.Fragment>
+                ) : null}
+              </ListItem>
+            )
+          );
+        })
+      )}
+    </List>
+  );
+}
```

Comment:
imo I find this block of code very difficult to read. This might just be because I'm not that familiar with lodash methods, so It's not a huge deal, but could we maybe simplify this by using the ES6 `reduce` method?

Reply:
@0Calories 🤔 it's a convoluted piece of code, yes! Which part do you think would work better with `.reduce`? I worry it'll end up more complicated. `.reduce` would only replace the `groupBy` logic, and it's a lot more verbose. Maybe I'm not seeing what you mean? I still need to do nested grouping and flattening at some point. If I try to `.reduce` the `errors` array, I'd need to do a whole bunch of error checking to see if certain keys exist or not, which will result in even more code

Reply:
You're right, reduce was just the first thing that came to mind tbh haha

I haven't thought too hard about how to simplify the code here, just wanted to throw ideas out there. If it would require too much effort then it's better to leave it as it is currently

Reply:
Totally fair!